presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestJsonDateTime.java

```diff
@@ -0,0 +1,199 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.json;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.prestosql.plugin.kafka.KafkaColumnHandle;
+import io.prestosql.plugin.kafka.encoder.EncoderColumnHandle;
+import io.prestosql.plugin.kafka.encoder.RowEncoder;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.type.DateType;
+import io.prestosql.spi.type.TimeType;
+import io.prestosql.spi.type.TimeWithTimeZoneType;
+import io.prestosql.spi.type.TimestampType;
+import io.prestosql.spi.type.TimestampWithTimeZoneType;
+import io.prestosql.spi.type.Type;
+import io.prestosql.testing.TestingConnectorSession;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+import static io.prestosql.spi.type.DateTimeEncoding.packDateTimeWithZone;
+import static io.prestosql.testing.assertions.Assert.assertEquals;
+
+public class TestJsonDateTime
+{
+    private static final JsonRowEncoderFactory ENCODER_FACTORY = new JsonRowEncoderFactory(new ObjectMapper());
+
+    private static EncoderColumnHandle createColumnHandle(String name, Type type, String dataFormat)
+    {
+        return createColumnHandle(name, type, dataFormat, null);
+    }
+
+    private static EncoderColumnHandle createColumnHandle(String name, Type type, String dataFormat, String formatHint)
+    {
+        return new KafkaColumnHandle(name, type, name, dataFormat, formatHint, false, false, false);
+    }
+
+    private static RowEncoder createRowEncoder(EncoderColumnHandle... columnHandles)
+    {
+        return ENCODER_FACTORY.create(TestingConnectorSession.SESSION, Optional.empty(), ImmutableList.copyOf(columnHandles));
+    }
+
+    @Test
+    public void testCustomFormat()
+    {
+        EncoderColumnHandle col1 = createColumnHandle("col1", DateType.DATE, "custom-date-time", "yyyy-dd-MM");
+        EncoderColumnHandle col2 = createColumnHandle("col2", TimeType.TIME, "custom-date-time", "kk:mm:ss.SSS");
+        EncoderColumnHandle col3 = createColumnHandle("col3", TimeWithTimeZoneType.TIME_WITH_TIME_ZONE, "custom-date-time", "kk:mm:ss.SSS Z");
+        EncoderColumnHandle col4 = createColumnHandle("col4", TimestampType.createTimestampType(3), "custom-date-time", "yyyy-dd-MM kk:mm:ss.SSS");
+        EncoderColumnHandle col5 = createColumnHandle("col5", TimestampWithTimeZoneType.createTimestampWithTimeZoneType(3), "custom-date-time", "yyyy-dd-MM kk:mm:ss.SSS Z");
+
+        RowEncoder rowEncoder = createRowEncoder(col1, col2, col3, col4, col5);
+
+        Block dateBlock = DateType.DATE.createBlockBuilder(null, 1).writeInt(18458);
```

Comment:
TBH, I find this test difficult to read.  I am looking at the setup and I wouldn't be able to predict what the outcome should be (not without writing a small program).  Maybe I'm missing something, but I don't think it adds anything to the tests in `TestKafkaIntegrationSmokeTest`.  If you wanted to see the full JSON message that is generated, you can get that in the round-trip test too (in `_message`).

Reply:
I wrote these tests for development because they are a million times faster than the roundtrip test in `TestKafkaIntegrationSmokeTest`. They both serve the same purpose now, idk if it would be a good idea to keep them around for development or if I should just use the roundtrip tests.